src/devices/cpu/sh/sh7604.h

```diff
@@ -0,0 +1,283 @@
+	uint16_t vcra_r();
+	void vcra_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint16_t vcrb_r();
+	void vcrb_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint16_t vcrc_r();
+	void vcrc_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint16_t vcrd_r();
+	void vcrd_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint16_t vcrwdt_r();
+	void vcrwdt_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+	uint32_t vcrdiv_r();
+	void vcrdiv_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint16_t intc_icr_r();
+	void intc_icr_w(offs_t offset, uint16_t data, uint16_t mem_mask = ~0);
+
+	// DIVU
+	uint32_t dvsr_r();
+	void dvsr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t dvdnt_r();
+	void dvdnt_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t dvdnth_r();
+	void dvdnth_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+	uint32_t dvdntl_r();
+	void dvdntl_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+
+	uint32_t dvcr_r();
+	void dvcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0);
+
+	// DMAC
+	template <int Channel> uint32_t vcrdma_r()
+	{
+		return m_vcrdma[Channel] & 0x7f;
+	}
+
+	template <int Channel> void vcrdma_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0)
+	{
+		COMBINE_DATA(&m_vcrdma[Channel]);
+		m_irq_vector.dmac[Channel] = m_vcrdma[Channel] & 0x7f;
+		sh2_recalc_irq();
+	}
+
+	template <int Channel> uint8_t drcr_r() { return m_dmac[Channel].drcr & 3; }
+	template <int Channel> void drcr_w(uint8_t data) { m_dmac[Channel].drcr = data & 3; sh2_recalc_irq(); }
+	template <int Channel> uint32_t sar_r() { return m_dmac[Channel].sar; }
+	template <int Channel> void sar_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0) { COMBINE_DATA(&m_dmac[Channel].sar); }
+	template <int Channel> uint32_t dar_r() { return m_dmac[Channel].dar; }
+	template <int Channel> void dar_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0) { COMBINE_DATA(&m_dmac[Channel].dar); }
+	template <int Channel> uint32_t dmac_tcr_r() { return m_dmac[Channel].tcr; }
+	template <int Channel> void dmac_tcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0) { COMBINE_DATA(&m_dmac[Channel].tcr); m_dmac[Channel].tcr &= 0xffffff; }
+	template <int Channel> uint32_t chcr_r() { return m_dmac[Channel].chcr; }
+	template <int Channel> void chcr_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0)
+	{
+		uint32_t old;
+		old = m_dmac[Channel].chcr;
+		COMBINE_DATA(&m_dmac[Channel].chcr);
+		m_dmac[Channel].chcr = (data & ~2) | (old & m_dmac[Channel].chcr & 2);
+		sh2_dmac_check(Channel);
+	}
+	uint32_t dmaor_r() { return m_dmaor & 0xf; }
+	void dmaor_w(offs_t offset, uint32_t data, uint32_t mem_mask = ~0)
+	{
+		if(ACCESSING_BITS_0_7)
+		{
+			uint8_t old;
+			old = m_dmaor & 0xf;
+			m_dmaor = (data & ~6) | (old & m_dmaor & 6); // TODO: should this be old & data & 6? bug?
+			sh2_dmac_check(0);
+			sh2_dmac_check(1);
+		}
+	}
```

Comment:
Do all these helpers need to be inlined here?  Are they only used in sh7604.cpp?  If they’re only used there, they can be moved to that file.  If they need to be instantiated from elsewhere, can they be moved to a .ipp file?

Reply:
They are only used by SH7604 and I don't think they really need to be in the header. I can move them out into sh6704.cpp.